src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509SecurityTokenProvider.cs

```diff
@@ -34,9 +34,9 @@ internal X509SecurityTokenProvider(X509Certificate2 certificate, bool clone)
             }
         }
 
-        protected override async Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
+        protected override Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
         {
-            return await Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
+            return Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
```

Comment:
@mconnew Are you sure? If we left out the type parameter, the compiler would infer a type of `Task<X509SecurityToken>`, which is different from the return type of `Task<SecurityToken>`.